spec.bs

```diff
@@ -3631,6 +3631,13 @@ The <dfn for=Navigator method>updateAdInterestGroups()</dfn> method steps are:
           set |ig|'s [=interest group/trusted bidding signals keys=] to |value|.
         1. Otherwise, jump to the step labeled <i><a href=#abort-update>Abort update</a></i>.
 
+        <dt>"`userBiddingSignals`"
+        <dd>
+        1. If |value| is a [=string=],
+          set |ig|'s [=interest group/user bidding signals=] to the result of [=parse a JSON

```

Comment:
The only problem here is that [parse a JSON string to an Infra value](https://infra.spec.whatwg.org/#parse-a-json-string-to-an-infra-value) expects to be called within a valid ES realm (since it can throw errors, and it operates on JS objects). Unfortunately, this algorithm mostly runs in parallel ([while at least one time, it does not](https://wicg.github.io/turtledove/#ref-for-interest-group-update)). That doesn't seem great, although I realize this algorithm already has this bug from several lines above where we already call this algorithm.

This is in contrast with the Chromium implementation, which uses `JSONStringValueSerializer` which is vacuumed-off logic isolated from v8 and everything. I believe @jyasskin has raised this issue other times in this specification (maybe looking into a realm-agnostic method of doing the same thing), but I can't find any bugs on this spec, Web IDL, or Infra that represents this. Jeffrey, do you think we could benefit from JSON algorithms that aren't tied to a realm? In the meantime, I'm wondering if we should file a bug on this spec, and tag each instance of this with that bug for tracking.

Reply:
Creating a bug for this and noting it everywhere we parse JSON [=in parallel=] SGTM.

Reply:
Infra should really let us do this: https://github.com/whatwg/infra/issues/625.

Reply:
@jyasskin had [a comment](https://github.com/WICG/turtledove/pull/601#discussion_r1230241193) about we cannot create JS objects [=in parallel=]. For parsing JSON to infra value, "[=parse a JSON string to an Infra value=] involve creating JS objects also, but because those objects don't escape, it's clear how to implement it without actually creating them".

So we only moved [=parsing JSON bytes to a JavaScript value=] to algorithms that are within single realm, but kept [=parse a JSON string to an Infra value=] in algorithms that run [=in parallel=]. But a "realm-agnostic method" to parse json would be great.

Reply:
Thanks for filing that, I couldn't remember if you had already filed something like that or not. OK I think we should make an `Issue: ` block next to this algorithm's usage and point to https://github.com/whatwg/infra/issues/625 from there.